lms/djangoapps/instructor_task/api.py

```diff
@@ -524,12 +524,14 @@ def generate_certificates_for_students(request, course_key, student_set=None, sp
     task_key = ""
     instructor_task = submit_task(request, task_type, task_class, course_key, task_input, task_key)
 
-    CertificateGenerationHistory.objects.create(
-        course_id=course_key,
-        generated_by=request.user,
-        instructor_task=instructor_task,
-        is_regeneration=False
-    )
+    cert_filter_args = {
+        "course_id": course_key,
+        "generated_by": request.user,
+        "instructor_task": instructor_task,
+        "is_regeneration": False
+    }
+
+    create_or_update_certificate_generation_history(**cert_filter_args).save()
```

Comment:
save() method is redundant
```suggestion
    create_or_update_certificate_generation_history(**cert_filter_args)
```

Reply:
Updated!